hub/powertoys/command-palette/publish-extension.md

```diff
@@ -1,72 +1,585 @@
+; 2. Replace EXTENSION_NAME with your extension name (e.g., CmdPalMyExtension)
+; 3. Replace DISPLAY_NAME with your extension's display name (e.g., My Extension)
+; 4. Replace DEVELOPER_NAME with your name (e.g., Your Name Here)
+; 5. Replace CLSID-HERE with extensions CLSID
+; 6. Update the default version to match your project file
+
+#define AppVersion "0.0.1.0"
+
+[Setup]
+AppId={{GUID-HERE}}
+AppName=DISPLAY_NAME
+AppVersion={#AppVersion}
+AppPublisher=DEVELOPER_NAME
+DefaultDirName={autopf}\EXTENSION_NAME
+OutputDir=bin\Release\installer
+OutputBaseFilename=EXTENSION_NAME-Setup-{#AppVersion}
+Compression=lzma
+SolidCompression=yes
+MinVersion=10.0.19041
+
+[Languages]
+Name: "english"; MessagesFile: "compiler:Default.isl"
+
+[Files]
+Source: "bin\Release\win-x64\publish\*"; DestDir: "{app}"; Flags: ignoreversion recursesubdirs
+
+[Icons]
+Name: "{group}\DISPLAY_NAME"; Filename: "{app}\EXTENSION_NAME.exe"
+
+[Registry]
+Root: HKCU; Subkey: "SOFTWARE\Classes\CLSID\{{CLSID-HERE}}"; ValueData: "EXTENSION_NAME"
+Root: HKCU; Subkey: "SOFTWARE\Classes\CLSID\{{CLSID-HERE}}\LocalServer32"; ValueData: "{app}\EXTENSION_NAME.exe -RegisterProcessAsComServer"
+```
+
+> [!TIP]
+> You can test this locally by having [.NET 9](https://dotnet.microsoft.com/download/dotnet/9.0) and [Inno Setup](https://jrsoftware.org/isdl.php) installed.
+>
+> ```powershell
+> # verify .Net 9 is installed
+> dotnet --version
+> 
+> # verify Inno Setup is installed
+> Test-Path "${env:ProgramFiles(x86)}\Inno Setup 6\iscc.exe"
+>
+> # build installer, this will take a while
+> .\build-exe.ps1 -Version "0.0.1.0"
+>
+> # verify that <ExtensionName>-Setup-0.0.1.0.exe is listed
+> Get-ChildItem "bin\Release\installer\" -File
+> ```
+
+### Automate with GitHub Actions
+
+> [!NOTE]
+> **What are GitHub Actions?**
+> GitHub Actions is a CI/CD platform that automates software workflows directly in your GitHub repository. For Command Palette extensions, GitHub Actions can automatically build your installer whenever you push code changes, create releases, and even submit updates to WinGet - eliminating manual build steps and ensuring consistent, reproducible builds.
+
+Now we'll set up GitHub Actions to automate the build and release process:
+
+1. `cd ..` up a directory, you should be in the directory that contains `<ExtensionName>.sln`
+1. create a new repo:
+
+```powershell
+mkdir .github/workflows
 ```
 
-You can run this from the "Run commands" command in Command Palette, or from the command-line, or from the Run dialog. 
+1. In the `workflows` directory, create a new file called `release-extension.yml`.
+1. Add the following content to the new file:
+
+```yml
+# TEMPLATE: Extension EXE Installer Build and Release Workflow
+# 
+# To use this template for a new extension:
+# 1. Copy this file to a new workflow file (e.g., release-myextension-exe.yml)
+# 2. Replace all instances of DEVELOPER_NAME with your developer name (e.g., Your Name Here)
```

Comment:
Is this step necessary? Don't see any other occurrences of `DEVELOPER_NAME` in the file besides this comment.

```suggestion
```